Source/EmptyDefaultValueProvider.cs

```diff
@@ -92,27 +92,32 @@ private static object GetReferenceTypeDefault(Type valueType)
 			else if (valueType == typeof(Task))
 			{
 				// Task<T> inherits from Task, so just return Task<bool>
-				return GetCompletedTaskForType(typeof(bool));
+				return Task.FromResult(false);
 			}
-			else if (valueType.GetTypeInfo().IsGenericType && valueType.GetGenericTypeDefinition() == typeof(IEnumerable<>))
+			else if (valueType.GetTypeInfo().IsGenericType)
```

Comment:
(This optimization is not directly related to this PR, but since one commit targets code de-duplication in `MockDefaultValueProvider`, it seems only logical to make the same effort in `EmptyDefaultValueProvider`, too.)